jsBasics/homeWork9/Book.js

```diff
@@ -0,0 +1,95 @@
+export default class Book {
+    constructor(title, author, yearOfPublication) {
+        this._title = title
+        this._author = author
+        this._yearOfPublication = yearOfPublication
+    }
+
+    get title(){
+        return this._title
+    }
+
+    set title(value){
+        if(typeof value !== "string"){
+            console.error("Error: Title should be type of string")
+            return
+        }
+
+        if(value.length < 1){
+            console.error("Error: Incorrect title value! Title length should be at least 1 symbol.")
+            return
+
+        }
+
+        this._title = value
+    }
+
+    get author(){
+        return this._author
+    }
+
+    set author(value){
+        if(typeof value !== "string"){
+            console.error("Error: Name of author should be type of string")
+            return
+        }
+
+        if(value.length < 2){
+            console.error("Error:Incorrect value! The name of author length should be at least 2 symbol.")
+            return
+        }
+
+        this._author = value
+    }
+
+    get yearOfPublication(){
+        return this._yearOfPublication
+    }
+
+    set yearOfPublication(value){
+        if(typeof value !== "number"){
+            console.error("Error: Year of publication should be type of number.")
+            return
+        }
+
+        if(value.length === 0){
+            console.error("Error: Empty year of publication value!")
+            return
+
+        }
+
+        if (value < 1800 || value > 2024) {
+            console.error("Error: Incorrect year of publication.")
+            return
+        }
+
+        this._yearOfPublication = value
+    }
+
+    printBookInfo(){
+        console.log("-------------------------------------------------");
+        console.log("Title the book : ", this._title);
+        console.log("Author : ", this._author);
+        console.log("Year of publication :", this._yearOfPublication);
+
+
+    }
+
+    static findOldestBook(books) {
+        if (!Array.isArray(books) || books.length === 0) {
+            console.error("Error: Invalid input or empty array.");
+            return
+        }
+
+        let oldestBook = books[0];
+
+        for (let i = 1; i < books.length; i++) {
+            if (books[i].yearOfPublication < oldestBook.yearOfPublication) {
+                oldestBook = books[i];
+            }
+        }
+
+        return oldestBook;
```

Comment:
you can make it simpler by using sort or reduce methods

Reply:
fixed
https://github.com/HelensBadCode/QA_Automation_JavaScript/pull/14/commits/ce8f1aa3acf59df20c33bd103360191993c4f521#diff-604cbc7b1cdfa47437bba93a6cee866c21c6e3ecd761f17c3b798a58e4049dbe

https://github.com/HelensBadCode/QA_Automation_JavaScript/pull/14/commits/ce8f1aa3acf59df20c33bd103360191993c4f521#diff-6052c4322ebefb98384a07362f9ba6b70cae650b87253e9a67e3d4635a147341